src/Game.js

```diff
@@ -40,217 +36,266 @@ class Game extends Component {
     return spritesPos;
   }
 
-  updateMove({ x, y }) {
-    if (this.spritesPos[y] === x) {
-      this.spritesPos[y] = -1;
-      this.remainingSprites -= 1;
-    }
+  const updateMove = useCallback(
+    ({ x, y }) => {
+      if (spritesPos[y] === x) {
+        spritesPos[y] = -1;
+        remainingSprites -= 1;
+      }
 
-    const hasFinished = this.remainingSprites === 0;
+      const hasFinishedAfterMove = remainingSprites === 0;
 
-    if (!this.state.hasFinished && hasFinished) {
-      this.setState({
-        hasFinished: true
-      });
-    }
-  }
+      if (!hasFinished && hasFinishedAfterMove) {
+        setHasFinished(true);
+      }
+    },
+    [spritesPos]
+  );
 
-  moveUp = ({ userPos }) => {
-    const userPosY = userPos.y;
-    let newY;
+  const moveUp = useCallback(
```

Comment:
Instead of adding `useCallback` here, you can change the setMoves usage to this

```js
setMoves(moves => moves + 1)
```

Reply:
Oh.. Okay! I'll change it